File: apps/pumpkin-net-models/Models/IHtmlBlock.cs
using System.Text.Json.Serialization;

namespace pumpkin_net_models.Models;

public interface IHtmlBlock
{
    [JsonPropertyName("id")]
    string? Id { get; set; }

    [JsonPropertyName("name")]
    string? Name { get; set; }

    [JsonPropertyName("styleKey")]
    string? StyleKey { get; set; }

    [JsonPropertyName("enabled")]
    bool Enabled { get; set; }

    [JsonPropertyName("schemaVersion")]
    int? SchemaVersion { get; set; }

    [JsonPropertyName("type")]
    string Type { get; set; }

    [JsonPropertyName("content")]
    object Content { get; set; }
}

File: packages/pumpkin-ts-models/src/models/IHtmlBlock.ts
/**
 * Base interface for all HTML blocks in Pumpkin CMS
 */
export interface IHtmlBlock {
  id?: string;
  name?: string;
  enabled?: boolean;
  schemaVersion?: number;
  type: string;
  content: Record<string, any>;
}

/**
 * Generic HTML block for unknown or custom block types
 */
export interface GenericHtmlBlock extends IHtmlBlock {
  type: string;
  content: Record<string, any>;
}
